### PR TITLE
#428-add-support-to-pr:reviewer:approved-event

### PR DESCRIPTION
**Pull request by MayaPetter**
adding support to reviewer approval event to update the build

### Your checklist for this pull request

- [x ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [ x] Ensure that the pull request title represents the desired changelog entry
- [ x] added SERVER_PULL_REQUEST_APPROVED event to NativeServerPullRequestHookProcessor.java and WebhookConfiguration.java
- [x ] https://github.com/jenkinsci/bitbucket-branch-source-plugin/issues/428 
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ x] Did you provide a test-case? That demonstrates feature works or fixes the issue.- 
   tested locally on a multibranch job, triggered scanning on or approval on bitbucket server

### CODE REVIEW

**Review by MayaPetter — COMMENTED** [review]
@tandibar @lucaspanjer  @cris

**Review by wahammed — APPROVED** [review]

**Comment by wahammed** [comment]
Who can merge this?

**Review by bitwiseman — APPROVED** [review]

**Comment by bitwiseman** [comment]
@wahammed 
There was a question on #428 as to whether this applies to Cloud.  I think it does.  
https://support.atlassian.com/bitbucket-cloud/docs/event-payloads/#EventPayloads-Approved

File: src/test/resources/com/cloudbees/jenkins/plugins/bitbucket/client/payload/2.0-repositories-amuniz-test-repos-pullrequests-1-commits_fields_values.hash_2Cvalues.author.raw_2Cvalues.date_2Cvalues.message_pagelen_1.json [hunk 1]
```diff
@@ -1,10 +0,0 @@
-{
-  "values": [{
-    "date": "2018-09-21T14:53:12+00:00",
-    "message": "[CI] Release version 1.0.0",
-    "hash": "bf0e8b7962c024026ad01ae09d3a11732e26c0d4",
-    "author": {
-      "raw": "Builder <no-reply@acme.com>"
-    }
-  }]
-}
```
Comment (bitwiseman) [inline src/test/resources/com/cloudbees/jenkins/plugins/bitbucket/client/payload/2.0-repositories-amuniz-test-repos-pullrequests-1-commits_fields_values.hash_2Cvalues.author.raw_2Cvalues.date_2Cvalues.message_pagelen_1.json:10]: Why was this deleted? I assume this was deleted to make a test pass?

**Review by bitwiseman — CHANGES_REQUESTED** [review]
This event is also present in Bitbucket Cloud. 
https://support.atlassian.com/bitbucket-cloud/docs/event-payloads/#Approved

And it was added to this codebase: 
https://github.com/jenkinsci/bitbucket-branch-source-plugin/blob/master/src/main/java/com/cloudbees/jenkins/plugins/bitbucket/hooks/HookEventType.java#L62

But then it is never used. 😭 

Please update this PR to add the Cloud version as well.

File: src/test/resources/com/cloudbees/jenkins/plugins/bitbucket/client/payload/2.0-repositories-amuniz-test-repos-pullrequests-1-commits_fields_values.hash_2Cvalues.author.raw_2Cvalues.date_2Cvalues.message_pagelen_1.json (hunk 1 above)
Comment (MayaPetter) [reply src/test/resources/com/cloudbees/jenkins/plugins/bitbucket/client/payload/2.0-repositories-amuniz-test-repos-pullrequests-1-commits_fields_values.hash_2Cvalues.author.raw_2Cvalues.date_2Cvalues.message_pagelen_1.json:10]: Hi, it was probably a mistake, I didn't try to delete it...

**Review by kdubrovnyi — APPROVED** [review]

**Comment by jkurek1** [comment]
Hi,
Any progress on that feature? It would be really good to have it.

**Comment by nfalco79** [comment]
Closing for the same reason of PR #827
